include/dfu/mcuboot_boot_mode.h

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_MCUBOOT_BOOT_MODE_H_
+
+#define BOOT_MODE_REQ_NORMAL		0x00
+#define BOOT_MODE_REQ_RECOVERY_DFU	0x01
```

Comment:
It's been a while since I've looked at zephyr's MCUboot related APIs, but I'm confused about this mix of "MCUBOOT" and "BOOT" in the namespace. Is this meant to be MCUboot-specific, or generic across all bootloaders?

If it is meant to be mcuboot-specific, is there any harm in using MCUBOOT consistently? (I think this is the real question I am trying to ask.)

If it is meant to be generic, why is MCUBOOT appearing sometimes here?